### PR TITLE
Use outline for bevel

### PR DESCRIPTION
**Pull request by sam-b-rose**
Uses an outline and outline offset for a more stable shadow bevel. A possible fix for https://github.com/Shopify/polaris-internal/issues/1095

Only updates the Banner and Card to first test, review, and refine.

Follows strategy tested in [this codepen](https://codepen.io/alex-page/pen/xxMyLXW)

<details>
<summary>Card example</summary>
Screenshot 2023-12-05 at 9 43 13 PM
</details>

<details>
<summary>Banner example</summary>
Screenshot 2023-12-05 at 9 43 02 PM
</details>

### CODE REVIEW

**Review by alex-page — APPROVED** [review]
Thats a nice clean up. CI looks to be a bit grumpy though.

File: polaris-tokens/src/shadows.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const shadows = {
+  100: '0px 1px 0px 0px rgba(26, 26, 26, 0.07)',
+  200: '0px 3px 1px -1px rgba(26, 26, 26, 0.07)',
+  300: '0px 4px 6px -2px rgba(26, 26, 26, 0.20)',
+  400: '0px 8px 16px -4px rgba(26, 26, 26, 0.22)',
+  500: '0px 12px 20px -8px rgba(26, 26, 26, 0.24)',
+  600: '0px 20px 20px -8px rgba(26, 26, 26, 0.28)',
+  bevel: '0px 1px 0px 0px rgba(194, 194, 194, 1)',
```
Comment (alex-page) [inline polaris-tokens/src/shadows.ts:8]: These get converted to `rem` right?

Comment (aaronccasanova) [reply polaris-tokens/src/shadows.ts:8]: Correct. Here's are some reference links:
- https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/themes/base/index.ts#L15-L22
- https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/utils.ts#L226-L254

Additionally, the same process is applied to variant themes:
https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/themes/utils.ts#L19-L44

File: polaris-tokens/src/themes/base/color.ts [hunk 2]
```diff
@@ -144,7 +144,9 @@ export type ColorBorderAlias =
   /** Specialty and component border colors. */
   | 'input-border-active'
   | 'input-border-hover'
-  | 'input-border';
+  | 'input-border'
+  | 'bevel-border'
+  | 'bevel-transparent-border';
```
Comment (aaronccasanova) [inline polaris-tokens/src/themes/base/color.ts:149]: Looking at other `color-<subgroup>` tokens, the naming convention puts `-transparent` after the subgroup name e.g. `color-bg-surface-transparent` and `color-bg-fill-transparent`. Should this be `color-bevel-border-transparent`?

File: polaris-tokens/src/shadows.ts (hunk 1 above)
Comment (aaronccasanova) [reply polaris-tokens/src/shadows.ts:8]: Looking at the usages, it should. Here are some reference links:
- https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/themes/base/index.ts#L15-L22
- https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/utils.ts#L226-L254

Additionally, the same process is applied to variant themes:
https://github.com/Shopify/polaris/blob/c64444dc0bb375779561af6411923acb27d7adef/polaris-tokens/src/themes/utils.ts#L19-L44

File: polaris-react/src/components/Box/Box.tsx [hunk 3]
```diff
@@ -270,7 +273,10 @@ export const Box = forwardRef<HTMLElement, BoxProps>(
       '--pc-box-max-width': maxWidth,
       '--pc-box-outline-color': outlineColor
         ? `var(--p-color-${outlineColor})`
-        : undefined,
+        : 'transparent',
+      '--pc-box-outline-offset': outlineOffset?.startsWith('-')
+        ? `calc(-1 * var(--p-border-width-${outlineOffset.slice(1)}))`
+        : `var(--p-border-width-${outlineOffset})`,
```
Comment (aaronccasanova) [inline polaris-react/src/components/Box/Box.tsx:279]: Heads up, we missed the `undefined` case when jamming on this solution e.g. the above would apply `var(--p-border-width-undefined)` as an inline style

**Comment by sam-b-rose** [comment]
Outline does not follow border radius in old versions of Safari.

Closing out this PR since it doesn't meet our minimum browser support 🥲

**Pull request closed without merging**